cookbooks/aws-parallelcluster-config/files/default/custom_action_executor/custom_action_executor.py

```diff
@@ -164,7 +267,12 @@ async def _execute_script(self, exe_script: ExecutableScript):
         try:
             # arguments are provided by the user who has the privilege to create/update the cluster
             subprocess.run(
-                [exe_script.path] + (exe_script.args or []), check=True, stderr=subprocess.PIPE
+                [exe_script.path] + (exe_script.args or []),
+                check=True,
+                stderr=subprocess.PIPE,
+                stdout=stdout,
+                text=True,
+                env=self._env_enricher.build_env(exe_script),
```

Comment:
Would it be possible to isolate the calls to subprocess.run to a single method so that we only need to #nosec in a single place? Though maybe this isn't a great suggestion as it could obfuscate the security risk, so I'm not sure.

Reply:
I do not agree, the different calls have a different of parameters and also the security risk changes, only this one accepts external inputs while putting the two together will make this less obvious IMHO